pkg/config/v2/config_writer.go

```diff
@@ -261,50 +261,44 @@ func toTopLevelConfigDefinition(context *serializerContext, configs []Config) (t
 }
 
 func extractConfigType(context *serializerContext, config Config) (typeDefinition, error) {
-	if config.Type.IsSettings() {
-		return getConfigTypeSettings(config, context)
-	} else if config.Type.IsEntities() {
-		return getConfigTypeEntities(config)
-	} else {
-		return getConfigTypeApi(config)
-	}
 
-}
+	switch t := config.Type.(type) {
+	case SettingsType:
+		scopeParam, found := config.Parameters[ScopeParameter]
```

Comment:
nitpick: This switch might be a bit easier to read if at least getConfigTypeSettings was still an extra method (e.g. one just taking context, parameters and type t)

Reply:
I extracted the part of getting the scope into its own method, IMO that makes the most sense.